### ISSUE TITLE
DataGrid .getPersistData() makes editing broken

### ISSUE THREAD

**Issue by ninopetrovic** [syncfusion/ej2-angular-ui-components#39]
```
public getGridPersistData() {
    const persistData = this.gridObj.getPersistData();
    this.persistData = JSON.parse(persistData);
    console.log('getPersistData', JSON.parse(persistData));
 }
```

After i call this function and I double-click on grid row to edit it I get this error:
![image](https://user-images.githubusercontent.com/43232121/61035264-279e0e80-a3c7-11e9-8dab-7e8c4a0cc91d.png)

If I comment the row: this.gridObj.getPersistData(); everything seems to work fine.

**Comment by thavasianand** [comment]
Hi ninopetrovic,

We have validated the provided information and checked with our end but we are not able to reproduce the reported problem. We have prepared a sample for your reference.

Sample : https://stackblitz.com/edit/angular-3aqxw4?file=app.component.ts

Could you please provide more information or below details that will helpful for us to validate further.

•	Did you used edit template in Grid ?
•	Share the code example and package version details to validate further. 

Regards,
Thavasianand S.

**Comment by ninopetrovic** [comment]
Hello, 

Yes I am using my own edit template on one of my columns which also uses my own component.
It is very weird because I fixed the problem like this: 
```
public getGridPersistData() {
    const persistData = this.gridObj.getPersistData();
    this.persistData = JSON.parse(persistData);
    this.gridObj.setProperties(this.persistData);
    console.log('getPersistData', JSON.parse(persistData));
  }
```
So looks like the persist data is actually fine.. just my grid somehow stops working when i get it out.

**Comment by ninopetrovic** [comment]
Update: yeah not even this fix that i posted in my last comment is viable since it makes columns totally broken...

before getPersistData():
![image](https://user-images.githubusercontent.com/43232121/61535782-64a37a00-aa33-11e9-8b99-a24897ce593b.png)
after getPersistData():
![image](https://user-images.githubusercontent.com/43232121/61535829-81d84880-aa33-11e9-8c3c-6421020a8c15.png)

The point is that i don't really need whole columns and dataSources nor Templates of all columns in my persistData... i only need the width.. reorder.. and visibility... basically when I'll load the persistData ill have all the columns already.. i just need to know how they were manipulated by the user.

**Comment by Sakthi7692** [comment]
Hi ninopetrovic,

We have validated the provided information and we suspect that in your application you are using old grid package version. We have already fixed the problem so we suggest you to upgrade to the latest version to resolve the problem.

Sample : https://stackblitz.com/edit/angular-lr3nam-bupze5?file=app.component.ts

Help documentation: https://ej2.syncfusion.com/angular/documentation/common/how-to/update-npm-package/

Release Notes: https://ej2.syncfusion.com/angular/documentation/release-notes/17.1.40/#grid

If you still facing the problem then share the package.json(code example) and if possible share the sample to validate further.

--Sakthi
